### PR TITLE
fix(FX-3586): Keep recent searches if log in to the same account

### PR DESCRIPTION
**Pull request by anastasiapyzhik**
The type of this PR is: **Bugfix**

This PR resolves [FX-3586]

### Description

Currently if user is logged out and back into the app they can't see their recent searches.
This PR keeps recent searches in async storage after log out and only removes them if previous user and newly logged in one are not the same.

#### Demo

https://user-images.githubusercontent.com/44819355/142632808-7954e2e0-bb7c-4bfa-b497-500d12b62cf6.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- keep recent searches after log out - anastasiapyzhik

</details>

[FX-3586]: https://artsyproduct.atlassian.net/browse/FX-3586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (keep recent searches after log out - anastasiapyzhik)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c7c13f320a91e94a44657329fd75f0fd795848d8
</p>

File: src/lib/Scenes/Home/Home.tests.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ describe(HomeQueryRenderer, () => {
     })
 
     expect(tree.root.findAllByType(SalesRailFragmentContainer)).toHaveLength(1)
-    expect(tree.root.findAllByType(FairsRailFragmentContainer)).toHaveLength(0)
+    expect(tree.root.findAllByType(FairsRailFragmentContainer)).toHaveLength(1)
```
Comment (dimatretyak) [inline src/lib/Scenes/Home/Home.tests.tsx:65]: Please make a rebase master of the branch. This has been fixed already

**Review by dimatretyak — DISMISSED** [review]
LGTM 👍 

https://user-images.githubusercontent.com/3513494/142673297-77248e59-5d49-40e6-8ca0-cd110ee99353.mp4

**Comment by MounirDhahri** [comment]
Sorry for the late review, this looks good to me

**Pull request merged**